File: examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/messages/CredDataHolder.java
package me.connect.sdk.java.sample.messages;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.Iterator;

import me.connect.sdk.java.message.Message;

public class CredDataHolder {
    public String id;
    public String name;
    public String attributes;
    public String offer;

    public CredDataHolder(String id, String name, String attributes, String offer) {
        this.id = id;
        this.name = name;
        this.attributes = attributes;
        this.offer = offer;
    }

    public static CredDataHolder extractDataFromCredentialsOfferMessage(Message msg) {
        try {
            JSONObject data = new JSONArray(msg.getPayload()).getJSONObject(0);
            String id = data.getString("claim_id");
            String name = data.getString("claim_name");
            JSONObject attributesJson = data.getJSONObject("credential_attrs");
            StringBuilder attributes = new StringBuilder();
            Iterator<String> keys = attributesJson.keys();
            while (keys.hasNext()) {
                String key = keys.next();
                String value = attributesJson.getString(key);
                attributes.append(String.format("%s: %s\n", key, value));
            }
            return new CredDataHolder(id, name, attributes.toString(), msg.getPayload());
        } catch (JSONException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: Examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/messages/CredDataHolder.kt
package me.connect.sdk.java.samplekt.messages

import me.connect.sdk.java.message.Message
import org.json.JSONArray
import org.json.JSONException


class CredDataHolder(var id: String, var name: String, var attributes: String, var offer: String) {
    companion object {
        fun extractDataFromCredentialsOfferMessage(msg: Message): CredDataHolder? {
            return try {
                val data = JSONArray(msg.payload).getJSONObject(0)
                val id = data.getString("claim_id")
                val name = data.getString("claim_name")
                val attributesJson = data.getJSONObject("credential_attrs")
                val attributes = StringBuilder()
                val keys = attributesJson.keys()
                while (keys.hasNext()) {
                    val key = keys.next()
                    val value = attributesJson.getString(key)
                    attributes.append(String.format("%s: %s\n", key, value))
                }
                CredDataHolder(id, name, attributes.toString(), msg.payload)
            } catch (e: JSONException) {
                e.printStackTrace()
                null
            }
        }
    }
}